Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -2682,8 +2730,114 @@ namespace OpenInfraPlatform
+					carve::math::Matrix inversePlacement = GeomUtils::computeInverse(placement);
+					carve::geom::vector<3> directionFromCenterToPoint = inversePlacement * point - carve::geom::VECTOR(0., 0., 0.);
+					// if the radial vector from Center to a point has coordinates (a,b), then the direction vector at that point is (−b,a)
+					return carve::geom::VECTOR(-directionFromCenterToPoint.y, directionFromCenterToPoint.x, 0.)
+				}
+
+				/*! \brief Converts \c IfcParameterValue to a angle parameter and passes it to getDirectionOfCurve.
+				* \param[in] circle					A pointer to data from \c IfcCircle.
+				* \param[in] parameter				A pointer to data from \c IfcParameterValue.
+				* \return							The Angle in radians.
+				* \note
+				*/
+				template<>
+				carve::geom::vector<3> getDirectionOfCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcCircle>& circle,
+					const typename IfcEntityTypesT::IfcParameterValue& parameter) const noexcept(false)
+				{
+					return getDirectionOfCurve(circle, parameter * this->UnitConvert()->getAngleInRadianFactor());
+				}
+
+				/*! \brief Calculates an angle of the circle.
+				* \param[in] circle					A pointer to data from \c IfcCircle.
+				* \param[in] angle					The angle.
+				* \return							The Angle in radians.
+				* \note
+				*/
+				template<>
+				carve::geom::vector<3> getDirectionOfCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcCircle>& circle,
+					const double& angle) const noexcept(false)
+				{
+					// https://math.stackexchange.com/questions/2680855/find-direction-of-tangent-vector-based-on-trajectory-along-circle
+					// if the radial vector from Center to a point has coordinates (a,b), then the direction vector at that point is (−b,a) 
+					return carve::geom::VECTOR(-sin(angle), cos(angle), 0.);
+				}
+
+
+				/**********************************************************************************************/
+				/*! \brief Converts \c IfcCartesianPoint to a angle parameter and passes it to getDirectionOfCurve.
+				* \param[in] ellipse				A pointer to data from \c IfcEllipse.
+				* \param[in] point					A pointer to data from \c IfcCartesianPoint.
+				* \return							The Angle in radians.
+				*/
+				template<>
+				carve::geom::vector<3> getDirectionOfCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcEllipse>& ellipse,
+					const typename IfcEntityTypesT::IfcCartesianPoint& cartesianPoint) const noexcept(false)
+				{
+					carve::geom::vector<3> point = placementConverter->convertIfcCartesianPoint(cartesianPoint);
+					carve::math::Matrix placement = placementConverter->convertIfcAxis2Placement(ellipse->Position);
+					carve::math::Matrix inversePlacement = GeomUtils::computeInverse(placement);
+					carve::geom::vector<3> directionFromCenterToPoint = inversePlacement * point - carve::geom::VECTOR(0., 0., 0.);
+					// if the radial vector from Center to a point has coordinates (a,b), then the direction vector at that point is (−b,a)

```

Comment:
this isn't necessarily true for ellipses. can you point me to the documentation, where you read this?

Reply:
I will look into it, I  was doing similarly to circle, but probably there should be other solution to this.

Reply:
I guess I have found a solution for this part. You can see my solution here 4d16c86

Reply:
It seems plausible.